bundles/framework/announcements/plugin/AnnouncementsPlugin.js

```diff
@@ -0,0 +1,242 @@
+            icon.addClass('icon-arrow-white-down');
+            div.append(me.announcementsContent);
+        },
+
+        /**
+        * @method closeSelection
+        * Programmatically closes the plugins interface as if user had clicked it close
+        */
+        closeSelection: function (el) {
+            var me = this;
+            me.open = false;
+            var element = el || me.getElement();
+            if (!element) {
+                return;
+            }
+            var icon = element.find('div.announcements-header div.announcements-header-icon');
+
+            icon.removeClass('icon-arrow-white-down');
+            icon.addClass('icon-arrow-white-right');
+            if (element.find('.content')[0]) {
+                element.find('.content').detach();
+            }
+        },
+
+        /**
+         * @private @method _createControlElement
+         * Constructs/initializes the indexmap  control for the map.
+         *
+         * @return {jQuery} element
+         */
+        _createControlElement: function () {
+            var me = this,
+                el = me.templates.main.clone(),
+                header = el.find('div.announcements-header');
+
+            header.append(me._loc.plugin.title);
+            me._bindHeader(header);
+            return el;
+        },
+
+        _bindAnnButton: function (ann, content) {
+            ann.on('click', function () {
+                this.classList.toggle('active');
+                if (content[0].style.maxHeight) {
+                    content[0].style.visibility = 'hidden';
+                    content[0].style.padding = null;
+                    content[0].style.maxHeight = null;
+                } else {
+                    content[0].style.visibility = 'visible';
+                    content[0].style.padding = '5px';
+                    content[0].style.maxHeight = content[0].scrollHeight + 'px';
+                }
+            });
+        },
+
+        _bindHeader: function (header) {
+            var me = this;
+            header.on('click', function () {
+                if (me.open) {
+                    me.closeSelection();
+                } else {
+                    me.openSelection();
+                }
+            });
+        },
+
+        /**
+        * @method addAnnouncements
+        * Adds announcements to the list from config
+        */
+        addAnnouncements: function () {
+            var me = this;
+            var announcementsIds = me._config.announcements;
+
+            var announcements = [];
+
+            for (const i of announcementsIds) {
+                for (const j of me.allAnnouncements) {
+                    if (j.id === i) {
+                        announcements.push(j);
+                    }
+                }
+            }
+
+            if (!me.open) {
+                delete me.announcementsContent;
+            } else {
+                jQuery('div.announcements-content').children().remove();
```

Comment:
This seems dangerous. Find a reference to the plugin UI instead of searching through the whole DOM.